src/main/java/nl/knaw/dans/ingest/DdIngestFlowApplication.java

```diff
@@ -219,14 +220,55 @@ public void run(final DdIngestFlowConfiguration configuration, final Environment
         environment.jersey().register(new CsvMessageBodyWriter());
     }
 
+    private static DepositToDvDatasetMetadataMapperFactory getMetadataMapperFactory(DdIngestFlowConfiguration configuration, DataverseClient dataverseClient) {
+        final var ingestFlowConfig = configuration.getIngestFlow();
+        return new DepositToDvDatasetMetadataMapperFactory(
+            ingestFlowConfig.getIso1ToDataverseLanguage(),
+            ingestFlowConfig.getIso2ToDataverseLanguage(),
+            ingestFlowConfig.getSpatialCoverageCountryTerms(),
+            dataverseClient
+        );
+    }
+
+    private DepositorAuthorizationValidator getDepositorAuthorizationValidator(DdIngestFlowConfiguration configuration, IngestAreaConfig autoIngestAreaConfig, DataverseClient dataverseClient) {
+        return buildDepositorAuthorizationValidator(
+            getDataverseService(configuration, dataverseClient),
+            configuration.getIngestFlow(),
+            autoIngestAreaConfig);
+    }
+
+    private static DataverseServiceImpl getDataverseService(DdIngestFlowConfiguration configuration, DataverseClient dataverseClient) {
```

Comment:
Any specific reason why you are returning the implementation and not the interface?